tests/django/test_model_mockers.py

```diff
@@ -0,0 +1,189 @@
+    model_mocker_factory,
+    schema_model_mockers_factory,
+    schema_models_factory,
+)
+from schematools.contrib.django.models import Dataset
+from schematools.types import DatasetTableSchema
+
+
+@pytest.mark.django_db
+def test_model_mocker_factory_registers_a_dynamic_model_in_the_app_config(afval_dataset) -> None:
+    """Prove that the model_mocker_factory registers the model as Django app."""
+    table_schema = afval_dataset.schema.tables[0]
+    assert isinstance(table_schema, DatasetTableSchema)
+    assert str(table_schema) == "<DatasetTableSchema: afvalwegingen.containers>"
+
+    # We bootstrap the DynamicModel inside the model_mocker_factory by calling the model_factory,
+    # which ends with app_config.register_model.
+    # So we assert that the DynamicModel is indeed registered.
+    model_mocker_factory(afval_dataset, table_schema, base_app_name="dso_api.dynamic_api")
+    registered_models = [f"{m._meta.app_label}.{m._meta.model_name}" for m in apps.get_models()]
+    assert "afvalwegingen.containers" in registered_models
+
+    # The afval_dataset has two tables (containers, clusters),
+    # so we check the second table (i.e. second model) too.
+    table_schema_with_id_as_string = afval_dataset.schema.tables[1]
+    assert isinstance(table_schema_with_id_as_string, DatasetTableSchema)
+    assert str(table_schema_with_id_as_string) == "<DatasetTableSchema: afvalwegingen.clusters>"
+
+    model_mocker_factory(
+        afval_dataset, table_schema_with_id_as_string, base_app_name="dso_api.dynamic_api"
+    )
+    registered_models_updated = [
+        f"{m._meta.app_label}.{m._meta.model_name}" for m in apps.get_models()
+    ]
+    assert "afvalwegingen.clusters" in registered_models_updated
+
+
+@pytest.mark.django_db
+def test_model_mocker_factory_sets_model_mocker_dataset_and_table_schema_through_dynamic_model(
+    afval_dataset,
+) -> None:
+    """Prove that the model_mocker_factory creates a linked DynamicModelMocker.
+
+    The mocker should be linked to a DynamicModel for a Dataset with a DatasetTableSchema
+    """
+    table_schema = afval_dataset.schema.tables[0]  # afvalwegingen.containers
+
+    model_mocker_cls = model_mocker_factory(
+        afval_dataset, table_schema, base_app_name="dso_api.dynamic_api"
+    )
+    # model_mocker_cls._meta is the DjangoModelFactory.Meta,
+    # so get_model_class() gives the Django Model for which
+    # the ModelFactory is implemented, which is the DynamicModel
+    # for the given AmsSchema dataset.table
+    AfvalwegingenContainersModel = model_mocker_cls._meta.get_model_class()
+    assert (
+        str(AfvalwegingenContainersModel)
+        == "<class 'dso_api.dynamic_api.afvalwegingen.models.containers'>"
+    )
+
+    assert type(AfvalwegingenContainersModel.get_dataset()) == Dataset
+    assert AfvalwegingenContainersModel.get_dataset() == afval_dataset
+    assert str(AfvalwegingenContainersModel.get_dataset()) == "afvalwegingen"
```

Comment:
Allemaal te vervangen door `assert model_mocker_cls._meta.get_model_class()` is AfvalwegingenContainersModel`

Reply:
Code is verwijderd. Test niet de mocking.